src/app/clients/ActionBar.tsx

```diff
@@ -0,0 +1,289 @@
+                            >
+                                {status}
+                            </MenuItem>
+                        );
+                    })}
+                </MenuList>
+            </Menu>
+            <Menu
+                open={actionAnchorEl !== null}
+                onClose={() => setActionAnchorEl(null)}
+                anchorEl={actionAnchorEl}
+            >
+                <MenuList id="action-menu">
+                    {actions.map((action, index) => {
+                        return (
+                            <MenuItem
+                                key={index}
+                                onClick={() => {
+                                    setSelectedAction(action);
+                                    setActionModal(true);
+                                    setActionAnchorEl(null);
+                                }}
+                            >
+                                {action}
+                            </MenuItem>
+                        );
+                    })}
+                </MenuList>
+            </Menu>
+            <OuterDiv>
+                <Button
+                    variant="contained"
+                    onClick={(event) => setStatusAnchorEl(event.currentTarget)}
+                    type="button"
+                    id="status-button"
+                >
+                    Statuses
+                </Button>
+                <Button
+                    variant="contained"
+                    onClick={(event) => setActionAnchorEl(event.currentTarget)}
+                    type="button"
+                    id="action-button"
+                >
+                    Actions
+                </Button>
+            </OuterDiv>
+        </>
+    );
+};
+
+interface SharedModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    data: Schema["parcels"][];
+    status: string | null;
+    onSubmit: (date: Dayjs) => void;
+    header: string;
+    headerId: string;
+    errorText: string | null;
+}
+
+const Row = styled.div`
+    display: flex;
+    gap: 1rem;
+    align-items: center;
+`;
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    align-items: stretch;
+`;
+
+const StatusText = styled.p`
+    margin-left: 1rem;
+    border-top: 1px solid darkgrey;
+    padding: 1rem 0;
+    &:last-child {
+        border-bottom: 1px solid darkgrey;
+    }
+`;
+
+const SharedModal: React.FC<SharedModalProps> = (props) => {
+    const [date, setDate] = useState(dayjs(Date()));
+
+    useEffect(() => {
+        setDate(dayjs(Date()));
+    }, [props.isOpen]);
+
+    return (
+        <Modal {...props} header={props.header} headerId={props.headerId}>
+            <ModalInner>
+                <Row>
+                    <p>Date: </p>
+                    <DatePicker
+                        value={date}
+                        defaultValue={date}
+                        onChange={(newDate) =>
```

Comment:
Is it worth putting these onChange function outside of the component? The function is quite big but its only used once so its up to you:)

Reply:
I would agree with this tbf